lib/connect/api.js

```diff
@@ -1,89 +1,73 @@
 'use strict'
 const cli = require('heroku-cli-util')
 const co = require('co')
-const https = require('https')
-const querystring = require('querystring')
+const axios = require('axios')
+
+function makeConnectClient (context) {
+  return axios.create({
+    headers: {
+      'Content-Type': 'application/json',
+      'Authorization': `Bearer ${context.auth.password}`,
+      'Heroku-Client': 'cli'
+    }
+  })
+}
 
-const regions = require('./regions.js')
+function makeDiscoveryClient (context) {
+  const baseURL = process.env['CONNECT_ADDON'] === 'connectqa'
+    ? 'https://hc-central-qa.herokai.com'
+    : 'https://hc-central.heroku.com'
+
+  return axios.create({
+    baseURL,
+    headers: {
+      'Content-Type': 'application/json',
+      'Authorization': `Bearer ${context.auth.password}`,
+      'Heroku-Client': 'cli'
+    }
+  })
+}
 
 exports.printResponseError = function (response) {
   console.log('Status code = ', response.statusCode, 'body = ', response.json)
 }
 
-let request = exports.request = function (context, method, path, data) {
-  let withBody = false
-  let headers = {
-    'Content-Type': 'application/json',
-    'Authorization': 'Bearer ' + context.auth.password,
-    'Heroku-Client': 'cli'
+let request = exports.request = function (context, method, url, data) {
+  if (!context.region) {
+    throw new Error('Must provide region URL')
   }
+  const connectClient = makeConnectClient(context)
+  return connectClient({
+    baseURL: context.region,
+    method,
+    url,
+    data
+  })
+}
 
-  // add data as querystring for GET request
-  if (data !== undefined && method === 'GET') {
-    path += ('?' + querystring.stringify(data))
-  }
-  // add data to body for POST/PUT requests
-  if (data !== undefined && ['POST', 'PUT', 'PATCH'].indexOf(method) !== -1) {
-    withBody = true
-    data = JSON.stringify(data)
-    headers['Content-Length'] = data.length
-  }
-
-  let hostname = regions.urlFromRegion(context.region)
-
-  let promise = new Promise(function (resolve, reject) {
-    let req = https.request({
-      hostname: hostname,
-      path: path,
-      method: method,
-      headers: headers
-    }, function (res) {
-      let body = ''
+let withUserConnections = exports.withUserConnections = co.wrap(function * (context, appName, flags, allowNone, heroku) {
```

Comment:
This whole thing is *soooo* much better, thank you so much for cleaning this up!